x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -191,51 +201,170 @@ public void authorize(final Authentication authentication, final String action,
+            runRequestInterceptors(requestInfo, authorizationInfo, engine, listener);
+        } else {
+            listener.onFailure(internalError("Only index actions can be authorized as child action"));
+        }
+    }
 
-            if (SystemUser.is(authentication.getUser())) {
-                // this never goes async so no need to wrap the listener
-                authorizeSystemUser(authentication, action, auditId, unwrappedRequest, listener);
+    private void authorizeStandaloneAction(RequestInfo requestInfo, String auditId, ActionListener<Void> listener) {
+        final ActionListener<AuthorizationInfo> authzInfoListener = wrapPreservingContext(ActionListener.wrap(authorizationInfo -> {
+            threadContext.putTransient(AUTHORIZATION_INFO_KEY, authorizationInfo);
+            maybeAuthorizeRunAs(requestInfo, auditId, authorizationInfo, listener);
+        }, listener::onFailure), threadContext);
+        getAuthorizationEngine(requestInfo.getAuthentication()).resolveAuthorizationInfo(requestInfo, authzInfoListener);
+    }
+
+    private String requireAuditId(Authentication authentication, String action, TransportRequest originalRequest) {
+        String auditId = AuditUtil.extractRequestId(threadContext);
+        if (auditId == null) {
+            // We would like to assert that there is an existing request-id, but if this is a system action, then that might not be
+            // true because the request-id is generated during authentication
+            if (isInternal(authentication.getUser())) {
+                auditId = AuditUtil.getOrGenerateRequestId(threadContext);
             } else {
-                final String finalAuditId = auditId;
-                final RequestInfo requestInfo = new RequestInfo(authentication, unwrappedRequest, action);
-                final ActionListener<AuthorizationInfo> authzInfoListener = wrapPreservingContext(ActionListener.wrap(
-                        authorizationInfo -> {
-                            threadContext.putTransient(AUTHORIZATION_INFO_KEY, authorizationInfo);
-                            maybeAuthorizeRunAs(requestInfo, finalAuditId, authorizationInfo, listener);
-                        }, listener::onFailure), threadContext);
-                getAuthorizationEngine(authentication).resolveAuthorizationInfo(requestInfo, authzInfoListener);
+                auditTrailService.get().tamperedRequest(null, authentication, action, originalRequest);
+                throw internalError(
+                    "Attempt to authorize action ["
+                        + action
+                        + "] for ["
+                        + authentication.getUser().principal()
+                        + "] without an existing request-id"
+                );
             }
         }
+        return auditId;
+    }
+
+    private ElasticsearchSecurityException internalError(String message) {
+        assert false : message;
```

Comment:
```suggestion
        // When running with assertions enabled (testing) kill the node so that there is a hard failure in CI
        assert false : message;
        // Otherwise (production) just throw an exception so that we don't authorize something incorrectly
```